target/bin/open-dkim

```diff
@@ -0,0 +1,194 @@
+#! /bin/bash
+
+KEYSIZE=4096
+SELECTOR=mail
+DOMAINS=
+
+function __usage
+{
+    echo -e "\e[35mOPEN-DKIM\e[31m(\e[93m8\e[31m)
+
+\e[38;5;214mNAME\e[39m
+    open-dkim - configure DomainKeys Identified Mail (DKIM)
+
+\e[38;5;214mSYNOPSIS\e[39m
+    ./setup.sh config dkim [ OPTIONS\e[31m...\e[39m ]
+
+\e[38;5;214mDESCRIPTION\e[39m
+    Configures DKIM keys. OPTIONS can be used to configure a more complex setup.
+    LDAP setups require these options.
+
+\e[38;5;214mOPTIONS\e[39m
+    \e[94mGeneric Program Information\e[39m
+        help       Print the usage information.
+
+    \e[94mConfiguration adjustments\e[39m
+        keysize    Set the size of the keys to be generated. Possible are 1024, 2024 and 4096 (default).
+        selector   Set a manual selector (default is 'mail') for the key. (\e[96mATTENTION\e[39m: NOT IMPLEMENTED YET!)
+        domains    Provide the domains for which keys are to be generated.
+
+\e[38;5;214mEXAMPLES\e[39m
+    \e[37m./setup.sh config dkim size 2048\e[39m
+        Creates keys of length 2048 bit in a default setup where domains are obtained from
+        your accounts.
+
+    \e[37m./setup.sh config dkim size 2048 selector 2021-dkim\e[39m
+        Creates keys of length 2048 bit in a default setup where domains are obtained from
+        your accounts. The DKIM selector used is '2021-dkim'.
+
+    \e[37m./setup.sh config dkim size 2048 selector 2021-dkim domain 'whoami.com,whoareyou.org'\e[39m
+        Appropriate for an LDAP setup. Creates keys of length 2048 bit in a default setup
+        where domains are obtained from your accounts. The DKIM selector used is '2021-dkim'.
+        The domains for which DKIM keys are generated are 'whoami.com' and 'whoareyou.org'.
+
+\e[38;5;214mEXIT STATUS\e[39m
+    Exit status is 0 if command was successful. If wrong arguments are provided or arguments contain
+    errors, the script will exit early with exit status 2.
+"
+}
+
+if [[ ${1:-} == 'help' ]]
+then
+  __usage
+  exit 0
+fi
+
+while [[ ${#} -gt 0 ]]
+do
+  case ${1} in
+    keysize )
+      if [[ -n ${2+'set'} ]]
+      then
+        KEYSIZE="${2}"
+        shift
+        shift
+      else
+        echo "No keysize provided after 'size' argument. Aborting." >&2
+        exit 2
+      fi
+      ;;
+
+    selector )
+      if [[ -n ${2+'set'} ]]
+      then
+        # shellcheck disable=SC2034
+        SELECTOR="${2}"
+        shift
+        shift
+      else
+        echo "No selector provided after 'selector' argument. Aborting." >&2
+        exit 2
+      fi
+      ;;
+
+    domain )
+      if [[ -n ${2+'set'} ]]
+      then
+        DOMAINS="${2}"
+        break
+        break
```

Comment:
Why are you using `break` here instead of `shift` ?

Reply:
Because I messed up. Can you correct this for me? Currently short on time...

Reply:
https://github.com/docker-mailserver/docker-mailserver/commit/6e6b5be1ee34fcce45448c1710f7d6a559bd1e54#diff-8ec5e15bee29bcdefebd50369ee7e19ad72d8b6e8c2d9475e3c4e8a463985a85

Reply:
https://github.com/docker-mailserver/docker-mailserver/runs/1929465064?check_suite_focus=true#step:6:325

@aendeavor any idea?:

```
# -- line differs --
# index    : 3
# expected :     open-dkim - configure DomainKeys Identified Mail (DKIM)
# actual   :     open-dkim - configure DomainKeys Identified Mail (DKIM)
# --
# 
```

Reply:
Actually your change (https://github.com/docker-mailserver/docker-mailserver/commit/d221c585c2adf9e08da72c440d21d0eb1b96bb00#diff-9f86d081884c7d659a2feaa0c55ad015a3bf4f1b2b0b822cd15d6c15b0f00a089) never went through CI as it was no PR and therefore tests didn't run only the linting.

Reply:
Might the output have problems with colors?

Reply:
Tested it locally with `\e[0m` but no difference. So the problem might be something else. But I don't get it. The output is exactly the same.

Reply:
@docker-mailserver/maintainers any idea?

Reply:
So I added --partial to the assert line and the test went through locally, will push this as a hotfix but I want to know which "invisible" char breaks the check

Reply:
This is absolutely weird... For `setup.sh` it seems to work just fine...